databroker/broker.py

```diff
@@ -1068,10 +1074,10 @@ def inner(stream):
 
 def header_io(db_in, db_out):
     def outer(f):
-        def inner(header):
+        def inner(headers):
```

Comment:
Fair enough, let's see where that goes.

I can merge this as is if you (1) rebase and (2) change `not isinstance(headers, list)` to `hasattr(headers, 'items')`. We should also headers to be any iterable collection, not strictly list. Looking for 'items' is a nice way to check if it's a dict (dict-like).